addon/webextension/sitehelper.js

```diff
@@ -0,0 +1,41 @@
+/* globals catcher, callBackground */
+/** This is a content script added to all Page Shot pages, and allows the site to
+    communicate with the add-on */
+
+window.sitehelper = (function () {
+
+  catcher.registerHandler((errorObj) => {
+    callBackground("reportError", errorObj);
+  });
+
+
+  function sendCustomEvent(name, detail) {
+    if (typeof detail == "object") {
```

Comment:
The type coercion here is a little sloppy, but objects in particular (as opposed to strings and numbers and null) trigger permission problems when transferring them between the add-on and content.  I forgot about null==object though, that's an annoying case, but at least undefined (as when the argument isn't given) will be left alone.